x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/InvalidateApiKeyRequestTests.java

```diff
@@ -69,7 +110,15 @@ public void writeTo(StreamOutput out) throws IOException {
                 super.writeTo(out);
                 out.writeOptionalString(realm);
                 out.writeOptionalString(user);
-                out.writeOptionalString(apiKeyId);
+                if (out.getVersion().onOrAfter(Version.V_7_10_0)) {
+                    if (Strings.hasText(apiKeyId)) {
+                        out.writeOptionalStringArray(new String[] { apiKeyId });
+                    } else {
+                        out.writeOptionalStringArray(null);
+                    }
+                } else {
+                    out.writeOptionalString(apiKeyId);
+                }
```

Comment:
Oh wow. I can't believe we need to have this. We definitely need to look at removing this class in a follow up PR.